<a href="https://colab.research.google.com/github/jajinkya/General-Loksabha-Election-of-India-2019/blob/master/Result_Of_Loksabha_Election.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Election data download 

In [1]:
from google.colab import files

uploaded = files.upload()

for fn in uploaded.keys():
  print('User uploaded file "{name}" with length {length} bytes'.format(
      name=fn, length=len(uploaded[fn])))

Saving Winning Candidate of 17th Lok Sabha.csv to Winning Candidate of 17th Lok Sabha (1).csv
User uploaded file "Winning Candidate of 17th Lok Sabha.csv" with length 40688 bytes


## Convert it into  Dataframe

In [2]:
import pandas as pd 
election = pd.read_csv("Winning Candidate of 17th Lok Sabha.csv")
election.head()

Sl. No.  ...           State
0        1  ...  Andhra Pradesh
1        2  ...  Andhra Pradesh
2        3  ...  Andhra Pradesh
3        4  ...  Andhra Pradesh
4        5  ...  Andhra Pradesh

[5 rows x 5 columns]

Number of rows and column in election dataframe.

In [3]:
election.shape

(543, 5)

In [4]:
# list of columns 
election.columns

Index(['Sl. No.', 'No. and Name of Parliamentary\rConstituency',
       'Name of the Elected\rMember', 'Party Affiliation\r(if any)', 'State'],
      dtype='object')

### Renaming the column names 

In [0]:
election.rename(columns={'No. and Name of Parliamentary\rConstituency':'Constituency','Name of the Elected\rMember':
                         'Winning_Candidate','Party Affiliation\r(if any)':'Party'}, inplace= True)

In [6]:
election.head()

Sl. No.  ...           State
0        1  ...  Andhra Pradesh
1        2  ...  Andhra Pradesh
2        3  ...  Andhra Pradesh
3        4  ...  Andhra Pradesh
4        5  ...  Andhra Pradesh

[5 rows x 5 columns]

## Data Cleaning

Have you noticed that in the Constituency column there is some string part of constituency name is unwanted? so we can remove it using ReGex expression. 

In [7]:
election.Constituency.replace(regex= True, inplace=True, to_replace= r'\d|-\r|\d|-', value='')
election.head()

Sl. No.  ...           State
0        1  ...  Andhra Pradesh
1        2  ...  Andhra Pradesh
2        3  ...  Andhra Pradesh
3        4  ...  Andhra Pradesh
4        5  ...  Andhra Pradesh

[5 rows x 5 columns]

Similarly using regex pattern we replace "\r" in party name with blank space. 

In [8]:
election.Party.replace(regex= True, inplace=True, to_replace= r'\r', value=' ')
election.head()

Sl. No.  ...           State
0        1  ...  Andhra Pradesh
1        2  ...  Andhra Pradesh
2        3  ...  Andhra Pradesh
3        4  ...  Andhra Pradesh
4        5  ...  Andhra Pradesh

[5 rows x 5 columns]

We can count ; how many constituencies  in India ?

In [9]:
election.Constituency.count()

543

In General Loksabha Election 2019 there are 543  constituencies from which MPs are elected in India. This number can be maximum 552. 

How many statewise constituencies are  there ?

In [0]:
state_constituency = election.groupby(by='State')


Here's the list of number of constituencies corresponding to state. 

In [11]:
list_ = []
for state in state_constituency.groups:
    list_2 = []
    list_2.append(state)
    list_2.append(state_constituency.get_group(state).Constituency.count())
    list_.append(list_2)

list_



[['Andaman & Nicobar Islands', 1],
 ['Andhra Pradesh', 25],
 ['Arunachal Pradesh', 2],
 ['Assam', 14],
 ['Bihar', 40],
 ['Chandigarh', 1],
 ['Chattisgarh', 11],
 ['Dadra & Nagar Haveli', 1],
 ['Daman & Diu', 1],
 ['Delhi', 7],
 ['Goa', 2],
 ['Gujrat', 26],
 ['Haryana', 10],
 ['Himachal Pradesh', 4],
 ['Jammu & Kashmir', 6],
 ['Jharkhand', 14],
 ['Karnataka', 28],
 ['Kerala', 20],
 ['Lakshadweep', 1],
 ['Madhya Pradesh', 29],
 ['Maharashtra', 48],
 ['Manipur', 2],
 ['Meghalaya', 2],
 ['Mizoram', 1],
 ['Nagaland', 1],
 ['Odisha', 21],
 ['Puducherry', 1],
 ['Punjab', 13],
 ['Rajasthan', 25],
 ['Sikkim', 1],
 ['Tamil Nadu', 39],
 ['Telangana', 17],
 ['Tripura', 2],
 ['Uttar Pradesh', 80],
 ['Uttarakhand', 5],
 ['West Bengal', 42]]

Let's abbreviate Parties name  and add as a another column in dataframe


In [0]:
list_ = []
for name in election.Party:
     first = ("".join([letter[:1].upper() for letter in name.split(' ')]))
     list_.append(first)

In [13]:
election['Symbol'] = list_
election.head()

Sl. No.   Constituency  ...           State Symbol
0        1     Araku (ST)  ...  Andhra Pradesh  YSRCP
1        2     Srikakulam  ...  Andhra Pradesh     TD
2        3   Vizianagaram  ...  Andhra Pradesh  YSRCP
3        4  Visakhapatnam  ...  Andhra Pradesh  YSRCP
4        5     Anakapalle  ...  Andhra Pradesh  YSRCP

[5 rows x 6 columns]

Lets count each parties winning seat in this election.

In [0]:
party_count = election.groupby(by ='Symbol')


In [15]:
stat = []
for name in party_count.groups:
    list_ = []
    list_.append(name)
    list_.append(party_count.get_group(name).Symbol.count())
    stat.append(list_)
    
    
stat

[['AAP', 1],
 ['ADS', 2],
 ['AIADMK', 1],
 ['AIMM', 2],
 ['AITC', 22],
 ['AIUDF', 1],
 ['AJSUP', 1],
 ['BJD', 12],
 ['BJP', 303],
 ['BSP', 10],
 ['CPOI', 2],
 ['CPOIM', 3],
 ['DMK', 23],
 ['I', 4],
 ['INC', 52],
 ['IUML', 3],
 ['J&KNC', 3],
 ['JDS', 1],
 ['JDU', 16],
 ['JMM', 1],
 ['KCM', 1],
 ['LJSP', 6],
 ['MNF', 1],
 ['NCP', 5],
 ['NDPP', 1],
 ['NPF', 1],
 ['NPP', 1],
 ['RLP', 1],
 ['RSP', 1],
 ['SAD', 2],
 ['SKM', 1],
 ['SP', 5],
 ['SS', 18],
 ['TD', 3],
 ['TRS', 9],
 ['V', 1],
 ['VCK', 1],
 ['YSRCP', 22]]

Here's the number of seats are won by party in each state.


In [16]:
data =  election.groupby(by=['State', 'Symbol'])
data.Symbol.count()

State                      Symbol
Andaman & Nicobar Islands  INC        1
Andhra Pradesh             TD         3
                           YSRCP     22
Arunachal Pradesh          BJP        2
Assam                      AIUDF      1
                           BJP        9
                           I          1
                           INC        3
Bihar                      BJP       17
                           INC        1
                           JDU       16
                           LJSP       6
Chandigarh                 BJP        1
Chattisgarh                BJP        9
                           INC        2
Dadra & Nagar Haveli       I          1
Daman & Diu                BJP        1
Delhi                      BJP        7
Goa                        BJP        1
                           INC        1
Gujrat                     BJP       26
Haryana                    BJP       10
Himachal Pradesh           BJP        4
Jammu & Kashmir            BJP        3
      